File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/Python/s826491904_commented.py
def main():
    N = int(input())
    max_val, min_val = map(int, input().split())
    for _ in range(1, N):
        in_val = int(input())
        max_val = max(max_val, in_val)
        min_val = min(min_val, in_val)
    print(max_val - min_val)

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/C++/s826491904_commented.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int N;
    cin >> N;
    int max_val, min_val;
    cin >> max_val;
    min_val = max_val;
    for(int i = 1; i < N; i++){
        int in;
        cin >> in;
        max_val = max(max_val, in);
        min_val = min(min_val, in);
    }
    cout << max_val - min_val << endl;
    return 0; // End of Code
}
